activation/activation_test.go

```diff
@@ -366,6 +369,7 @@ func TestBuilder_StartSmeshingAvoidsPanickingIfPrepareInitializerReturnsError(t
 	goroutineCount := runtime.NumGoroutine()
 	tab.StartSmeshing(tab.coinbase, PostSetupOpts{})
 	require.Equal(t, goroutineCount, runtime.NumGoroutine())
```

Comment:
imo counting the go routines isn't necessary if we call `eg.Wait()` in the test.

Reply:
If smeshing had been started `eg.wait()` would take a very long time to return. So i think the go-routine count is useful or we could timeout while waiting for `eg.wait()` to return.

Reply:
But smeshing never starts in this test, because `PrepareInitializer` returns an error and we are asserting that this error is propergated and returned by `tab.StartSmeshing`.

Imo this tests is sufficient to check that `StartSmeshing` doesn't start a smeshing session if `PrepareInitializer` fails:

```golang
func TestBuilder_StartSmeshing_SessionNotStartedOnFailPrepare(t *testing.T) {
	l := log.NewMockLogger(gomock.NewController(t))
	tab := newTestBuilder(t)
	tab.log = l

	// Stub these methods in case they get called
	tab.mpost.EXPECT().GenerateProof(gomock.Any(), gomock.Any()).AnyTimes().Return(&types.Post{}, &types.PostMetadata{}, nil)
	tab.mclock.EXPECT().AwaitLayer(gomock.Any()).AnyTimes()

	// Set expectations, we don't expect StartSession to be called, since
	// StartSmeshing should exit early if PrepareInitializer returns an error.
	testError := errors.New("avoid panic")
	tab.mpost.EXPECT().PrepareInitializer(gomock.Any(), gomock.Any()).Return(testError)

	require.ErrorIs(t, tab.StartSmeshing(tab.coinbase, PostSetupOpts{}), testError)
	tab.eg.Wait()
}
```

`eg.Wait()` should return almost immediately here and if for some reason it doesn't then this test should fail (in a sense that it will time out after whatever we have currently defined as test timeout).

Reply:
> If smeshing had been started `eg.wait()` would take a very long time to return. So i think the go-routine count is useful or we could timeout while waiting for `eg.wait()` to return.

Counting the go routines doesn't guarantee that eg.wait() doesn't block, afaik the runtime can start go routines any time (for instance to garbage collect) so the count might be off anyway.

Additionally if smeshing had been started the test would fail with a proper error message too (along the lines of `StartSmeshing` has unexpectedly been called), if it does immediately or after the test times out imo is not as important as making sure the test properly cleans up after itself when it passes 🤷

Reply:
@fasmat, I just want to unblock an annoyingly flaky test, not to re-implement it. TBH, I'd remove it completely and probably refactor the code to not panic in the first place.

Reply:
`eg.Wait()` getting stuck is not a bad thing imo. It signals that the test failed, and we have a test timeout. So if a test takes longer than 5 minutes to complete, test execution will be aborted.

Reply:
@poszu "re-implementing" is a bit much for splitting a test into 2 without changing a single line of code 😅 

first half:
![Screenshot 2023-06-20 at 11 50 20](https://github.com/spacemeshos/go-spacemesh/assets/5011972/33d0dc36-ffe5-423b-822e-ba3e62c3a4bb)

second half:
![Screenshot 2023-06-20 at 11 50 47](https://github.com/spacemeshos/go-spacemesh/assets/5011972/153d4504-c209-40e8-a1d0-7920a36b7130)